### ISSUE TITLE
Having build issues when trying to build from source

### ISSUE THREAD

**Issue by sahussain5599** [Project-OSRM/osrm-backend#7015]
# Issue
I am following the following steps to build the OSRM backend from source on my local system:

sudo apt install build-essential git cmake pkg-config \
libbz2-dev libxml2-dev libzip-dev libboost-all-dev \
lua5.2 liblua5.2-dev libtbb-dev

then,

mkdir -p build
cd build
cmake ..
cmake --build .
sudo cmake --build . --target install

as I run the command cmake --build .  I run into the following error. Bare in mind the error is few thousand lines but this is a small snapshot of it

![error](https://github.com/user-attachments/assets/2efdb1a3-142c-447f-8fd4-a979649a8d80)

This is being run on WSL and all the C++ versions and everything is fine apparently.

I would really appreciate if someone will be able to help me out with this. Cheers!

**Comment by woodpeck** [comment]
Confirm this happens on a plain Ubuntu 22.04 setup when following "building from source" instructions which claim to target Ubuntu 22.04. If this is no longer true and a different distribution, or a different version of Ubuntu, is now suggested for a clean build, it would be great if the instructions could be modified accordingly.

**Comment by somogyirolandhu** [comment]
Hello,
Make sure to check the appropriate repo after cloning.
Use git checkout v5.27.1 after git clone.

**Comment by dpbnasika** [comment]
yes version v5.27.1 build works fine, i had the similar issue with current master branch on Ubuntu 22.04.

**Comment by shin0859** [comment]
Maybe this will solve it

https://www.linuxbabe.com/ubuntu/install-osrm-ubuntu-22-04-open-source-routing-machine
AND
ｘ git clone https://github.com/Project-OSRM/osrm-backend.git
〇 git clone –branch v5.27.1 https://github.com/Project-OSRM/osrm-backend.git
